src/agri_ai/core/database_pool.py

```
@@ -0,0 +1,319 @@
+                self.database = self.client[self.settings.mongodb_database]
+                self.is_connected = True
+                self.last_health_check = datetime.now()
+                
+                logger.info(f"Successfully connected to MongoDB database: {self.settings.mongodb_database}")
+                
+            except (ConnectionFailure, ServerSelectionTimeoutError) as e:
+                logger.error(f"Failed to connect to MongoDB: {e}")
+                raise DatabaseConnectionError(
+                    f"MongoDB接続に失敗しました: {str(e)}",
+                    context={"mongodb_uri": self.settings.mongodb_uri}
+                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)